### DISCUSSION TITLE
Could not create runner: listen ip4:tcp : socket: protocol not supported

### DISCUSSION THREAD

**Discussion by tonyscy** [projectdiscovery/naabu#572] [Q&A] [answered]
```console
$ naabu -hc

Version: 2.1.1
Operative System: linux
Architecture: amd64
Go Version: go1.18.7
Compiler: gc
Privileged/NET_RAW: Ok
Config file "/root/.config/naabu/config.yaml" Read => Ok
Config file "/root/.config/naabu/config.yaml" Write => Ok
IPv4 connectivity to scanme.sh:80 => Ok
IPv6 connectivity to scanme.sh:80 => Ko (dial tcp6 [2400:6180:0:d0::91:1001]:80: connect: connection refused)
```

and i try use Version: 2.1.0 too。

it happend in wsl ！！

![image](https://user-images.githubusercontent.com/42638375/208617043-c19d09e3-b592-4e5e-98c2-8f8c7110f094.png)

**Comment by ehsandeep** [comment]
@tonyscy you mentioned wsl, to confirm, you are facing this issue in [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) environment?

**Comment by tonyscy** [comment]
Yes, I ran into this issue in wsl,








Edward&nbsp;
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "projectdiscovery/naabu"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年12月20日(星期二) 晚上7:45
***@***.***&gt;;
抄送:&nbsp;"Edward ***@***.******@***.***&gt;;
主题:&nbsp;Re: [projectdiscovery/naabu] Could not create runner: listen ip4:tcp : socket: protocol not supported (Issue #515)





 
@tonyscy you mentioned wsl, to confirm, you are facing this issue in WSL environment?

**Comment by Mzack9999** [comment, marked as answer]
Unfortunately, WSL doesn't fully support raw sockets, so under this environment, only connect scan is supported (the same happens with nmap)
